### PR TITLE
Fix few compilation warnings (shift)

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR fixes few compilation warnings.

### CODE REVIEW

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveE.cpp [hunk 1]
```diff
@@ -84,7 +84,7 @@ void FiniteDifferenceSolver::MacroscopicEvolveECartesian (
     // properties from their respective staggering to the Ex, Ey, Ez locations
     amrex::GpuArray<int, 3> const& sigma_stag = macroscopic_properties->sigma_IndexType;
     amrex::GpuArray<int, 3> const& epsilon_stag = macroscopic_properties->epsilon_IndexType;
-    amrex::GpuArray<int, 3> const& mu_stag = macroscopic_properties->mu_IndexType;
+    //amrex::GpuArray<int, 3> const& mu_stag = macroscopic_properties->mu_IndexType;
```
Comment (ax3l) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveE.cpp]: Let's remove the line, this is now defined in `Source/FieldSolver/FiniteDifferenceSolver/MacroscopicProperties/MacroscopicProperties.cpp` directly.
```suggestion
```

**Review by ax3l — APPROVED** [review]

File: Source/WarpX.cpp [hunk 2]
```diff
@@ -1246,8 +1246,8 @@ WarpX::LowerCornerWithGalilean (const Box& bx, const amrex::Array<amrex::Real,3>
 {
     amrex::Real cur_time = gett_new(lev);
     amrex::Real time_shift = (cur_time - time_of_last_gal_shift);
-    amrex::Array<amrex::Real,3> galilean_shift = { vv_galilean[0]*time_shift, vv_galilean[1]*time_shift, vv_galilean[2]*time_shift };
-    return WarpX::LowerCorner(bx, galilean_shift, lev);
+    amrex::Array<amrex::Real,3> t_galilean_shift = { vv_galilean[0]*time_shift, vv_galilean[1]*time_shift, vv_galilean[2]*time_shift };
```
Comment (LDAmorim) [inline Source/WarpX.cpp]: Hi @lucafedeli88 ,
I just noticed that you fix one of the compile time warnings regarding shadowing the variable `galilean_shift`.
So I was wondering if you would like to address the other similar warnings, for example:
```
./Source/FieldSolver/WarpXPushFieldsEM.cpp: In member function 'void WarpX::ApplyInverseVolumeScalingToCurrentDensity(amrex::MultiFab*, amrex::MultiFab*, amrex::MultiFab*, int)':
./Source/FieldSolver/WarpXPushFieldsEM.cpp:362:35: warning: declaration of 'galilean_shift' shadows a member of 'WarpX' [-Wshadow]
         std::array<amrex::Real,3> galilean_shift = {0,0,0};
```
and
```
In file included from ./Source/FieldSolver/WarpXPushFieldsEM.cpp:9:
./Source/WarpX.H:272:54: note: shadowed declaration is here
     amrex::Array<amrex::Real,3> galilean_shift = {{0}};
```
in this PR.
Or if you prefer I can try to do it.
I just think these are mainly your files and you know better how you wish to fix them.

ps: I found them when compiling with openPMD in CORI@NERSC.

Cheers,
Diana

**Review by LDAmorim — COMMENTED** [review]
Thank you for this PR @lucafedeli88 .
I just added a minor comment regarding compile time warnings fixes that you might want to include in this PR.

File: Source/WarpX.cpp (hunk 2 above)
Comment (ax3l) [reply Source/WarpX.cpp]: Sounds good, we could actually re-name the member variable from `galilean_shift` to `m_galilean_shift` to fix them all.

Comment (lucafedeli88) [reply Source/WarpX.cpp]: Hi @LDAmorim ! 
Thanks for the suggestion! I should have fixed this issue with my latest commits. I did as @ax3l suggested and I renamed `galilean_shift` to `m_galilean_shift` in WarpX class and I amended the other files accordingly.
I did the same also for the `v_galilean` member variable of `WarpXParticleContainer` and of `WarpX`, which is now `m_v_galilean`.

I've also fixed another warning related to a comparison between an `int` and an `unsigned int` .

**Pull request merged**